### PR TITLE
Refactor project to be a normal ROS package

### PR DESCRIPTION
**Pull request by felixdivo**
This was required for  #23 / https://github.com/ros/rosdistro/pull/39892.

It also switches from black to `ruff format` and activates two previously broken test cases. Also restores installability on CI for ROS Rolling.

### CODE REVIEW

**Comment by felixdivo** [comment]
I don't really understand how to fix the error here: https://github.com/felixdivo/ros2-easy-test/actions/runs/8659687762/job/23746089747?pr=33 ...

**Comment by Timple** [comment]
Perhaps because `extras_require` lists a lot of dependencies. These will (but shouldn't be) pip installed?

File: .github/workflows/python-package.yaml [hunk 1]
```diff
@@ -28,26 +28,30 @@ jobs:
       -
         name: Install dependencies and the package
         run: |
-          apt-get update -q && apt-get install -qy python3-pip python-is-python3 ros-$ROS_DISTRO-example-interfaces
+          apt-get update -q && apt-get install -qy python3-pip python-is-python3 python3-colcon-ros ros-$ROS_DISTRO-example-interfaces
           pip install --upgrade pip setuptools
-          python -m pip install -e '.[dev]'
+          python -m pip install -e './ros2_easy_test[dev]'
```
Comment (vik748) [inline .github/workflows/python-package.yaml:33]: Typically a ros package is installed or used by adding it in a workspace under the `src` directory.
Once the directory structure is established and dependencies are moved to package.xml, we should be able to run something like:
```
#Install dependencies
rosdep update
rosdep install --from-paths src -y --ignore-src --rosdistro humble

# Build
colcon build

# Source the package
source install/setup.bash

# Test
colcon test
```
Alternatively you could use a predefined action like: [ros-tooling/action-ros-ci@master](https://github.com/ros-tooling/action-ros-ci)

**Comment by felixdivo** [comment]
Well, pip is still required to install `makefun` (sadly not available via `package.xml`/Ros Index). But aside from that, installing it for normal usage works without it. Installing for development needs more dependencies, so there you'd still require pip.

**Pull request merged**